authority/provisioner/x5c_test.go

```diff
@@ -554,65 +509,245 @@ func TestX5C_AuthorizeSign(t *testing.T) {
 	}
 }
 
+func TestX5C_AuthorizeRevoke(t *testing.T) {
+	type test struct {
+		p     *X5C
+		token string
+		code  int
+		err   error
+	}
+	tests := map[string]func(*testing.T) test{
+		"fail/invalid-token": func(t *testing.T) test {
+			p, err := generateX5C(nil)
+			assert.FatalError(t, err)
+			return test{
+				p:     p,
+				token: "foo",
+				code:  http.StatusUnauthorized,
+				err:   errors.New("x5c.AuthorizeRevoke: x5c.authorizeToken; error parsing x5c token"),
+			}
+		},
+		"ok": func(t *testing.T) test {
+			certs, err := pemutil.ReadCertificateBundle("./testdata/certs/x5c-leaf.crt")
+			assert.FatalError(t, err)
+			jwk, err := jose.ParseKey("./testdata/secrets/x5c-leaf.key")
+			assert.FatalError(t, err)
+
+			p, err := generateX5C(nil)
+			assert.FatalError(t, err)
+			tok, err := generateToken("foo", p.GetName(), testAudiences.Revoke[0], "",
+				[]string{"test.smallstep.com"}, time.Now(), jwk,
+				withX5CHdr(certs))
+			assert.FatalError(t, err)
+			return test{
+				p:     p,
+				token: tok,
+			}
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			tc := tt(t)
+			if err := tc.p.AuthorizeRevoke(context.TODO(), tc.token); err != nil {
+				if assert.NotNil(t, tc.err) {
+					sc, ok := err.(errs.StatusCoder)
+					assert.Fatal(t, ok, "error does not implement StatusCoder interface")
+					assert.Equals(t, sc.StatusCode(), tc.code)
+					assert.HasPrefix(t, err.Error(), tc.err.Error())
+				}
+			} else {
+				assert.Nil(t, tc.err)
+			}
+		})
+	}
+}
+
+func TestX5C_AuthorizeRenew(t *testing.T) {
+	type test struct {
+		p    *X5C
+		code int
+		err  error
+	}
+	tests := map[string]func(*testing.T) test{
+		"fail/renew-disabled": func(t *testing.T) test {
+			p, err := generateX5C(nil)
+			assert.FatalError(t, err)
+			// disable renewal
+			disable := true
+			p.Claims = &Claims{DisableRenewal: &disable}
+			p.claimer, err = NewClaimer(p.Claims, globalProvisionerClaims)
+			assert.FatalError(t, err)
+			return test{
+				p:    p,
+				code: http.StatusUnauthorized,
+				err:  errors.Errorf("x5c.AuthorizeRenew; renew is disabled for x5c provisioner %s", p.GetID()),
+			}
+		},
+		"ok": func(t *testing.T) test {
+			p, err := generateX5C(nil)
+			assert.FatalError(t, err)
+			return test{
+				p: p,
+			}
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			tc := tt(t)
+			if err := tc.p.AuthorizeRenew(context.TODO(), nil); err != nil {
+				if assert.NotNil(t, tc.err) {
+					sc, ok := err.(errs.StatusCoder)
+					assert.Fatal(t, ok, "error does not implement StatusCoder interface")
+					assert.Equals(t, sc.StatusCode(), tc.code)
+					assert.HasPrefix(t, err.Error(), tc.err.Error())
+				}
+			} else {
+				assert.Nil(t, tc.err)
```

Comment:
Not sure this actually tests the AuthorizeRenew func. We're just checking that we didn't add an error to the test case.

Reply:
I'm more so checking that I didn't expect an error and then not receive one.

Reply:
Does something update tc.err durring the test? Wouldn't you want to assert.Nil(t, the_returned_error_if_any)?

Reply:
Nothing updates `tc.err`. `tc.err` is something I set in the test conditions so that if an error does appear then I can compare it to `tc.err` to make sure it's the error I expected. 

In this condition we already know that there was no returned error, so it doesn't make sense to assert that again - since in the first condition we asserted that the `err != nil`. But suppose the error was `nil` and I actually expected an error. That's the case I'm testing here. 

There are probably other ways to test this, I just found this most convenient.

Reply:
Oh so this is testing the "ok" case where you expect the test error to be nil. Now it makes sense!